crates/re_chunk/src/slice.rs

```diff
@@ -0,0 +1,453 @@
+use arrow2::array::{
+    Array as ArrowArray, BooleanArray as ArrowBooleanArray, ListArray,
+    PrimitiveArray as ArrowPrimitiveArray, StructArray,
+};
+
+use itertools::Itertools;
+use nohash_hasher::IntSet;
+use re_log_types::Timeline;
+use re_types_core::ComponentName;
+
+use crate::{Chunk, ChunkTimeline};
+
+// ---
+
+// NOTE: Not worth writing tests for all of these, until some subtle bug comes up.
+// Most of them are indirectly stressed by our higher-level query tests anyhow.
+
+impl Chunk {
+    /// Slices the [`Chunk`] vertically.
+    ///
+    /// The result is a new [`Chunk`] with the same columns and (potentially) less rows.
+    ///
+    /// This cannot fail nor panic: `index` and `len` will be capped so that they cannot
+    /// run out of bounds.
+    /// This can result in an empty [`Chunk`] being returned if the slice is completely OOB.
+    #[inline]
+    pub fn row_sliced(&self, index: usize, len: usize) -> Self {
+        let Self {
+            id,
+            entity_path,
+            heap_size_bytes: _,
+            is_sorted,
+            row_ids,
+            timelines,
+            components,
+        } = self;
+
+        // NOTE: Bound checking costs are completely dwarfed by everything else, and preventing the
+        // viewer from crashing is more important than anything else in any case.
+
+        if index >= self.num_rows() {
+            return self.emptied();
+        }
+
+        let end_offset = usize::min(index.saturating_add(len), self.num_rows());
+        let len = end_offset.saturating_sub(index);
+
+        if len == 0 {
+            return self.emptied();
+        }
+
+        let is_sorted = *is_sorted || (len < 2);
```

Comment:
This is yet another case where we can guarantee positives but not negatives: gotta take the safe route if `!is_sorted`.